hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -422,9 +422,22 @@ private void submitConnectToClusterTask() {
                             logger.warning("No connection to cluster: " + clusterId);
                         }
 
-                        submitConnectToClusterTask();
+                        submitConnectToClusterTask(initialConnection);
                     }
                 }
+                if (initialConnection && !activeConnections.isEmpty()) {
+                    // Send state to cluster for async clients in async way.
+                    executor.submit(() -> {
+                        try {
+                            client.sendStateToCluster();
+                        } catch (Throwable e) {
+                            logger.severe("Could not send state to cluster during async start. "
+                                    + "Client will shutdown because it cannot operate with state not sent to cluster."
+                                    + " Error: " + e.getMessage());
```

Comment:
```suggestion
                                    + "Client will shutdown because it cannot operate with state not sent to cluster.", e);
```
this would print the stacktraces etc., which might be more useful